### PR TITLE
NetworkManager 1.2 support

### PR DESCRIPTION
**Pull request by lkundrak**
The branch contains changes that are necessary for the properties plugin to continue working with upcoming nm-connection-editor 1.2 and more. The details are explained in the "nm: split the plugin into two halves" commit message.

Please note that I just did a buildcheck; I don't have a necessary setup to test this (= I'm probably a lazy fuck). Given I've done analogous changes to other plugins and they worked I wouldn't expect it to be too broken.

### CODE REVIEW

File: appdata/network-manager-sstp.appdata.xml.in [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright Lubomir Rintel 2016 -->
+<component type="addon">
+  <id>network-manager-sstp</id>
+  <project_license>GPL-2.0+</project_license>
+  <metadata_license>CC0-1.0</metadata_license>
```
Comment (enaess) [inline appdata/network-manager-sstp.appdata.xml.in]: Does that mean, you want to change the license to Creative Commons for redistribution?

**Comment by enaess** [comment]
Thanks a bunch Lubomir Rintel,

I've committed this change to the main branch (not sure if that was a good idea to take it verbatim). You want me to add you to the maintainers or credits section somewhere? I am about to get this building and tested.

- Eivind

File: appdata/network-manager-sstp.appdata.xml.in (hunk 1 above)
Comment (lkundrak) [inline appdata/network-manager-sstp.appdata.xml.in]: That is just the licence for the AppStream XML file. The distributors typically bundle these snippets in their app store streams and a permissive license just avoids hassle there.

The tag above is about the project; still GPL.

**Comment by enaess** [comment]
Hi Lubomir, 
I've made a few other changes mainly to the nm-sstp-service.conf and added the link dependency to nm-sstp-pppd-plugin.so that is required the additional object files for calling to get the password details. Smoke tested it and it seems like it worked fine. 

Regards,
Eivind

**Comment by szpak** [comment]
@enaess Do you plan any additional changes in the 1.2.0 branch before making a release?

**Comment by enaess** [comment]
Hi Marcin,
I need to smoke test it as my current test server is down, and then I'll need to figure out this Ubuntu package problem that was reported where nm-sstp-service.conf was installed in /usr/lib/NetworkManager/VPN and not under /etc. I've been working long hours at work, and haven't had much time for it otherwise. Will look at this today.
Regards,
- Eivind


 

    Den Lørdag, 21. mai 2016 5.28 skrev Marcin Zajączkowski <notifications@github.com>:
 

 @enaess Do you plan any additional changes in the 1.2.0 branch before making a release?—

**Comment by szpak** [comment]
No problem, take your time Eivind. I can use current Git snapshot to make a test build for Fedora and I will later upgrade it to the release version when available.

**Comment by enaess** [comment]
@szpak I did test the .deb file via launchpad and that seems to be working. What version of Fedora are you going to build this against? And would you need to make any changes to the rpmspec?

**Comment by szpak** [comment]
For 24 and 25 which has NM 1.2.x. Hopefully I will only need to adjust spec file to use Git snapshot.

Btw, in the meantime @lkundrak did some [enhancements](http://pkgs.fedoraproject.org/cgit/rpms/NetworkManager-sstp.git/log/) in a spec file and it would be probably good to backport them to your spec as well (one day when NM-sstp will be fully available in Fedora).

**Comment by enaess** [comment]
Cool!
I'll download a FC25 image, and give your .spec a spin. Let me know if you need any more merges.

- Eivind

**Comment by szpak** [comment]
@enaess Today I tried to build 1.2.0 Git snapshot in my CORP repo and I failed. The problem was sstp-client which was in the meantime removed from Fedora... What is a status of this project? Can the current version of sstp-client be used with NM-sstp 1.2.0? Do you plan to migrate it to GitHub?

@lkundrak Are you able to resurrect sstp-client in Fedora and take care of it?

**Comment by enaess** [comment]
I wasn't aware of that there was was a problem. How are you building this?

**Comment by szpak** [comment]
The issue with Fedora is not related to you. The previous Fedora maintainer of that package probably just orphaned it. My question was rather if you see any compatibility issues with NM-sstp after an upgrade to 1.2.0?

**Comment by szpak** [comment]
Btw, in the end I managed to build NM-sstp for Fedora 24 and rawhide in my CORP repo [1]. I will try to find someone to test it with some sstp client.

[1] - https://copr.fedorainfracloud.org/coprs/szpak/NetworkManager-sstp/build/362316/

**Comment by enaess** [comment]
I can give it a spin on a machine here, getting Fedora 24 installed in a VM.

**Comment by enaess** [comment]
@szpak You have any special instructions on how I can install your build?

**Comment by szpak** [comment]
> sudo dnf copr enable szpak/NetworkManager-sstp
> sudo dnf install NetworkManager-sstp-gnome

should be enough. However COPR currently returns Error 500 : Internal Server Error :(

**Comment by enaess** [comment]
Hi @szpak, I did setup a FC24 VM and as a result, I had to merge changes back into master and create a tag as @lkundrak asked me to. I may need to create a release branch for 1.2 for any other changes needed to get the build rolling for FC24. Currently I am stuck creating a rpm build with the files put in the right places (lib vs. lib64?), and the rpmspec file seems to reference 0.9.10 where it should say 1.2. You should be able to create a tarball at the 1.2 release tag using: 

git clone --branch release-1.2 <network-manager-git-repository>

**Comment by szpak** [comment]
You don't need to create an RPM manually. Have you tried to use the COPR repo I maintain (mentioned [above](https://github.com/enaess/network-manager-sstp/pull/11#issuecomment-228590636))?

In case you would like to do it the changes required to create an RPM from a Git snapshot are in my branch (based on @lkundrak work):
http://pkgs.fedoraproject.org/cgit/rpms/NetworkManager-sstp.git/log/?h=user/szpak/NetworkManager-1.2-git
(you can also just rebuild my SRPM file)

However it would be better to just merge Lubomir [changes](http://pkgs.fedoraproject.org/cgit/rpms/NetworkManager-sstp.git/log/?h=user/lkundrak/NetworkManager-1.2) into your branch release-1.2 and release NM-sstp-1.2. After that my changes would not be needed any longer.

Btw, one guy reported that he was able to use my NM-sstp package from CORP in Fedora 24 without any problem. Once I convince @lkundrak to take over sstp-client package in Fedora ;) I will be able to roll out NN-sstp 1.2 to Fedora repo (if released :) ).

**Comment by szpak** [comment]
Btw2, @enaess What happened to the release-1.2 branch? It seems to be removed. I like it as in the master there is [commit](https://github.com/enaess/network-manager-sstp/commit/13eba25b428af81789d74fe83b05d99e811ee65c) which seemed to be not compatible with 1.2.0.

**Comment by enaess** [comment]
For some bizarre reason, it looked like the release-1.2 branch in it's original incarnation only had 1 change different between them. I merged that change to the ubuntu/control file back into main and deleted the branch. The changes in master holds additional refactoring done in the original NetworkManager-pptp project for release 1.3 and above and would not be compatible with the FC-24 release. The tag I put in place for release-1.2 is at the appropriate change of master.

**Comment by enaess** [comment]
@szpak I have not tried building on COPR (assuming this is something similar to launchpad for Ubuntu), i.e. you submit your build job and it will create the packages for you. I've been wondering if both the .rpmspec and the Ubuntu related files should have been decoupled from the project and maintained by their individual distributors themselves, but oh well...

How do I use COPR? Do you have a link with quick start information or you able to teach me how in a few sentences?

**Comment by szpak** [comment]
> sudo dnf copr enable szpak/NetworkManager-sstp
> sudo dnf install NetworkManager-sstp-gnome

**Comment by szpak** [comment]
I understand with the tag. Regarding SPEC file it would have to be updated with Lubomir [changes](http://pkgs.fedoraproject.org/cgit/rpms/NetworkManager-sstp.git/log/?h=user/lkundrak/NetworkManager-1.2) to make it compatible with the version 1.2 in a tarball. However I don't know if anyone needs to make it's own RPM having it available in the Feodra repo. Maybe backporting changes from 1.2 release in Fedora to your master would be enough (as a reference)? The SPEC file will live its own life in the Fedora Git anyway.

**Comment by enaess** [comment]
Just created the branch v1.2 and updated the .rpm file and moved the tag release-1.2 to this branch. You will need to checkout that branch and run autogen.sh on your FC system to build the NetworkManager-1.2.0.tar.bz2 and copy it into rpmbuilds directory. Then run rpmbuild on the rpmspec file.

**Comment by szpak** [comment]
Looks good. Do you plan to release the official tarball of 1.2.0 available from SF (as usual)?

**Comment by enaess** [comment]
Sight, you really make me go all the way ... I did upload the source archive, the .deb files, and then finally the .rpm files. Though, people should really add the ppa on Ubuntu to make this work.

**Comment by szpak** [comment]
Great :)

Having RPM available in Fedora (soon) that RPM would not be needed.

Btw, why they don't want to make this package directly into Debian?

**Comment by enaess** [comment]
Great,
I'd love to be able to get this package directly into debian but have not been able to find a sponsor yet. Also, I've tried to push it up myself only to find it rejected like a few months later. Seems like people strangely enough still keep asking me for .rpm or .deb files

**Pull request merged**